Add @attributes as reserved (#592)
unicode-org/message-format-wg: spec/syntax.md

However, specifications or implementations defining _markup_ might impose requirements
on the pairing, ordering, or contents of _markup_ during _formatting_.

## Attributes

> An **_<dfn>attribute</dfn>_** is a key with an optional value that
> is attached to the _expression_ or _markup_ as a whole.

Reviewer: This is what we intend to use it for, but I would probably not say it like this. Note that `key` should be `identifier` and that "the expression" is slightly quirky. Perhaps:

```suggestion
An **_<dfn>attribute</dfn>_** is an identifier with an optional value
that appears in an _expression_ or in _markup_.
```

> _Attributes_ are reserved for use by future specification versions.

Reviewer: I would put this sentence first and perhaps in bold in its own paragraph? I would make it more emphatic because we give examples later and we don't want to give the wrong impression.

```suggestion
**_Attributes_ are reserved for standardization by future versions of this specification.**
Examples in this section are meant to be illustrative and
might not match future requirements or usage.
```

_Attributes_ are prefixed by a U+0040 COMMERCIAL AT `@` sign,
followed by an _identifier_.

> An _attribute_ MAY have a _value_ separated from the _identifier_

Reviewer: Add a couple of words for clarity.

```suggestion
An _attribute_ MAY have a _value_ which is separated from the _identifier_
```

However, specifications or implementations defining _markup_ might impose requirements
on the pairing, ordering, or contents of _markup_ during _formatting_.

## Attributes

> An **_<dfn>attribute</dfn>_** is an _identifier_ with an optional value
> that appears in an _expression_ or in _markup_.
> 
> **_Attributes_ are reserved for standardization by future versions of this specification.**
> Examples in this section are meant to be illustrative and
> might not match future requirements or usage.

Author: ```suggestion
**_Attributes_ are reserved for standardization by future versions of this specification.**
Examples in this section are meant to be illustrative and
might not match future requirements or usage.

An **_<dfn>attribute</dfn>_** is an _identifier_ with an optional value
that appears in an _expression_ or in _markup_.
```
